### ISSUE TITLE
Checking the type of "useHash" and "useHmac" shows an error.

### ISSUE THREAD

**Issue by tomatoaiu** [Drazail/react-native-hash#57]
**Describe the bug**
Using this library in a project using TypeScript will result in an error.

**To Reproduce**
1. `npm i -g @react-native-community/cli`
2. `npx react-native init MyApp --template react-native-template-typescript`
3. `cd MyApp`
4. `npm i react-native-hash --legacy-peer-deps`
5. Change `skipLibCheck` in tsconfig.json from `true` to `false`.
6. Rewrite App.tsx as follows.

```jsx
import React from 'react';
import {SafeAreaView, Text} from 'react-native';

import {useHash} from 'react-native-hash';

const App = () => {
  const hash = useHash();
  return (
    <SafeAreaView>
      <Text>{hash[0]}</Text>
    </SafeAreaView>
  );
};

export default App;
```

7. `npx tsc --noEmit`
8. An error is output.

```log
~/develop/MyApp ❯ npx tsc --noEmit
node_modules/react-native-hash/index.d.ts:68:3 - error TS1015: Parameter cannot have question mark and initializer.

68   hmacAlgo?: CONSTANTS.HashAlgorithms = CONSTANTS.HashAlgorithms.md5,
     ~~~~~~~~

node_modules/react-native-hash/index.d.ts:68:3 - error TS2371: A parameter initializer is only allowed in a function or constructor implementation.

68   hmacAlgo?: CONSTANTS.HashAlgorithms = CONSTANTS.HashAlgorithms.md5,
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-hash/index.d.ts:68:24 - error TS2724: '"/Users/user/develop/MyApp/node_modules/react-native-hash/index".CONSTANTS' has no exported member named 'HashAlgorithms'. Did you mean 'HmacAlgorithms'?

68   hmacAlgo?: CONSTANTS.HashAlgorithms = CONSTANTS.HashAlgorithms.md5,
                          ~~~~~~~~~~~~~~

node_modules/react-native-hash/index.d.ts:69:3 - error TS2371: A parameter initializer is only allowed in a function or constructor implementation.

69   initialMessage: ?string = "hello World"
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-hash/index.d.ts:69:19 - error TS8020: JSDoc types can only be used inside documentation comments.

69   initialMessage: ?string = "hello World"
                     ~~~~~~~

node_modules/react-native-hash/index.d.ts:77:3 - error TS1015: Parameter cannot have question mark and initializer.

77   hmacAlgo?: CONSTANTS.HmacAlgorithms = CONSTANTS.HmacAlgorithms.HmacMD5,
     ~~~~~~~~

node_modules/react-native-hash/index.d.ts:77:3 - error TS2371: A parameter initializer is only allowed in a function or constructor implementation.

77   hmacAlgo?: CONSTANTS.HmacAlgorithms = CONSTANTS.HmacAlgorithms.HmacMD5,
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-hash/index.d.ts:77:24 - error TS2724: '"/Users/user/develop/MyApp/node_modules/react-native-hash/index".CONSTANTS' has no exported member named 'HmacAlgorithms'. Did you mean 'HashAlgorithms'?

77   hmacAlgo?: CONSTANTS.HmacAlgorithms = CONSTANTS.HmacAlgorithms.HmacMD5,
                          ~~~~~~~~~~~~~~

node_modules/react-native-hash/index.d.ts:78:3 - error TS2371: A parameter initializer is only allowed in a function or constructor implementation.

78   initialMessage: ?string = "hello World",
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-hash/index.d.ts:78:19 - error TS8020: JSDoc types can only be used inside documentation comments.

78   initialMessage: ?string = "hello World",
                     ~~~~~~~

node_modules/react-native-hash/index.d.ts:79:3 - error TS2371: A parameter initializer is only allowed in a function or constructor implementation.

79   initialSecret: ?string = "SecretKey"
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/react-native-hash/index.d.ts:79:18 - error TS8020: JSDoc types can only be used inside documentation comments.

79   initialSecret: ?string = "SecretKey"
                    ~~~~~~~


Found 12 errors in the same file, starting at: node_modules/react-native-hash/index.d.ts
```

**Expected behavior**
I am hoping that using `npx tsc --noEmit` will not output an error.

**Screenshots**

**Desktop (please complete the following information):**
```log
System:
    OS: macOS 12.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 124.94 MB / 32.00 GB
    Shell: 3.3.1 - /opt/homebrew/bin/fish
  Binaries:
    Node: 16.13.2 - ~/.nodenv/versions/16.13.2/bin/node
    Yarn: Not Found
    npm: 8.5.5 - ~/.nodenv/versions/16.13.2/bin/npm
    Watchman: 2022.01.31.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 21, 22, 28, 30, 31, 32, 33
      Build Tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0
      System Images: android-21 | Intel x86 Atom_64, android-21 | Google APIs ARM EABI v7a, android-21 | Google APIs Intel x86 Atom, android-22 | ARM EABI v7a, android-30 | Google APIs ARM 64 v8a, android-31 | Google APIs ARM 64 v8a, android-31 | Google Play ARM 64 v8a, android-32 | Google APIs ARM 64 v8a, android-33 | Google APIs ARM 64 v8a, android-33 | Google Play ARM 64 v8a, android-33 | Google Play Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7935034
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 15.0.2 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.4 => 0.69.4
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

**Smartphone (please complete the following information):**

**Additional context**
I am wondering if the following changes to the "index.d.ts" of "react-native-hash" would prevent the error from being output.

https://github.com/Drazail/react-native-hash/pull/58

**Comment by Drazail** [comment]
@tomatoaiu 
Merged #58 into master, please install the module from github (npm install Drazail/react-native-hash#master) and let me know if the issue is resolved so we can push an update to npm.

**Comment by tomatoaiu** [comment]
@Drazail 
Thank you for your prompt reply review.
I was able to resolve the error successfully 👍

**Comment by tomatoaiu** [comment]
@Drazail 
I installed react-native-hash via Github, no problem!
Please release react-native-hash!🚀

**Comment by Drazail** [comment]
thank you for your contribution.
Published as [3.0.1](https://www.npmjs.com/package/react-native-hash/v/3.0.1)
